docs/user/interop.rst

```diff
@@ -197,15 +197,33 @@ One can declare it as follows in Scala Native:
 
     def test(f: unsafe.CFuncPtr1[CString, Unit]): Unit = unsafe.extern
 
-`CFuncPtrN` types are SAM (single abstract method) traits. You
-can define them by creating a class that inherits from the corresponding
-trait:
+`CFuncPtrN` types are final classes containing pointer to underlying
+C function pointer. They automatically handle boxing call arguments
+and unboxing result. You can create them from C pointer using `Ptr` helper methods:
 
 .. code-block:: scala
 
-   val myfuncptr = new unsafe.CFuncPtr0[Unit] {
-     def apply(): Unit = println("hi there!")
-   }
+    def fnDef(str: CString): CInt = ???
+
+    val anyPtr: Ptr[Byte] = Ptr.cFuncPtrToPtr {
+      CFuncPtr1.fromScalaFunction(fnDef)
+    }
```

Comment:
Is this still accurate?
```suggestion
    val anyPtr: Ptr[Byte] = CFuncPtr1.toPtr {
      CFuncPtr1.fromScalaFunction(fnDef)
    }
```
?

Perhaps the implicit conversion even kicks in now, so we could write:
```suggestion
    val anyPtr: Ptr[Byte] = CFuncPtr1.toPtr(fnDef(_))
```
?

Reply:
The second one would not work right now, because it should use `CFuncPtr.toPtr(_: CFuncPtr)`. For some reason, the compiler does not try to make implicit conversion here.  
Or maybe there is a cleaner way to write following, which would work 
```
CFuncPtr.toPtr {
      val x: CFuncPtr[CString, CInt] = fnDef _  //would be converted using implicit converison
      x
}
```